src/main/java/com/es/phoneshop/utils/ValidationPredicateUtil.java

```diff
@@ -19,12 +19,14 @@
 
     public static Predicate<String> getDatePredicate() {
         return getDefaultParamPredicate().and(s -> {
+            LocalDate parse;
             try {
-                LocalDate.parse(s);
+                parse = LocalDate.parse(s);
+                parse.isBefore(LocalDate.now());
             } catch (DateTimeParseException e) {
                 return false;
             }
-            return true;
+            return parse.isEqual(LocalDate.now()) || parse.isAfter(LocalDate.now());
```

Comment:
And honestly there is no much profit here from using isAfter() rather then compareTo(), despite what i said on the call, so you could use whatever you like. Just take look at implementation of LocalDate.compareTo(..) and LocalDate.isBefore(..) and you will see that difference is only in readability.